docs/dev/microtvm_design.rst

```diff
@@ -0,0 +1,340 @@
+can theoretically step back and the user can compile and run the code separately.
+
+However, for AutoTVM, TVM needs some automated flow to handle the following tasks:
+
+1. Integrate operator implementations, the TVM C Runtime library, and the TVM RPC Server library into the
+   firmware project containing user-supplied SoC Initialization.
+2. Build the resulting project.
+3. Program the built firmware onto a (specific) attached device.
+4. Identify the serial port or other transport to be used by TVM to drive remote execution.
+
+At present, TVM expects the user to supply an implementation of the ``tvm.micro.Compiler``,
+``tvm.micro.Flasher``, and ``tvm.micro.Transport`` interfaces. TVM then:
+
+1. Builds each piece separately as a library
+2. Builds the libraries into a binary firmware image.
+3. Programs the firmware image onto an attached device.
+4. Opens a serial port to serve as the RPC server transport.
+
+This design was chosen to reduce build times for microTVM (the common libraries need to be build
+only once per candidate operator implemmentation). In practice, these projects are extremely small
+and compile relatively quickly. Compared with the added complexity of this tighter build integration
+with TVM, the performance gains are likely not worth it. A future design will consolidate the build
+tasks into a single step and narrow the interface to provide a better integration.
+
+Measuring operator performance
+------------------------------
+
+The TVM C runtime depends on user-supplied functions to measure time on-device. Users should implement
+``TVMPlatformTimerStart`` and ``TVMPlatformTimerStop``. These functions should measure wall time, so there
+are some pitfalls in implementing this function:
+
+1. If the CPU could halt or sleep during a computation (i.e. if it is being done on an accelerator),
+   a cycle counter should likely not be used as these tend to stop counting while the CPU is asleep.
+2. The granularity of these functions can be relaxed as needed to extend the range of the timer
+   device. However, if granularity is too coarse, a sub-optimal schedule may be used.
+3. An error should be raised if the timer overflows.
+4. The timer should not interrupt computation unless absolutely necessary. Doing so may affect the
+   accuracy of the results.
+5. Calibrating the output against a wall clock is ideal, but it will likely be too cumbersome. A
+   future PR could enable some characterization of the platform timer by e.g. measuring the internal
+   oscillator against a reference such as an external crystal.
+
+Future Work
+===========
+
+Ahead-of-Time Runtime
+----------------------
+
+A limitation of the Graph Runtime is the amount of memory overhead required in parsing the JSON.
+The current implementation contributes to the dynamic memory usage of microTVM. An ahead-of-time
+runtime can avoid the need for any Graph JSON allocations and improve inference speed.
```

Comment:
I am not fully sure I understand the paragraphs, and not too sure how `Ahead-of-Time Runtime` could improve the speed. Maybe bit adding bit more explanation could help readers?

Reply:
I clarified a bit and added some additional explanation